lib/puppet/node/facts.rb

```diff
@@ -116,8 +116,25 @@ def add_timestamp
     @timestamp = Time.now
   end
 
+  def to_yaml
+    facts_to_display = Psych.parse_stream(YAML.dump(self))
```

Comment:
Will `Psych` always be the current YAML engine? I'm not 100% sure if it will when using JRuby/puppetserver.

Reply:
I've made some digging and i found out that Psych is the default engine for ruby since ruby 2.0.0 and that is also applicable for jruby (https://github.com/jruby/jruby/blob/master/lib/ruby/stdlib/yaml.rb).